services/movie_session.py

```diff
@@ -0,0 +1,50 @@
+from django.db.models import QuerySet
+from db.models import MovieSession
+from datetime import datetime
+
+
+def create_movie_session(
+    movie_show_time: datetime,
+    movie_id: int,
+    cinema_hall_id: int
+) -> MovieSession:
+    new_movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+    return new_movie_session
+
+
+def get_movies_sessions(
+    session_date: str = None
+) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+    session_id: int,
+    show_time: datetime = None,
+    movie_id: int = None,
+    cinema_hall_id: int = None
+) -> None:
+    update_ms = MovieSession.objects.filter(id=session_id)
+    update_ms.update()
+    if show_time:
+        update_ms.update(show_time=show_time)
+    if movie_id:
+        update_ms.update(movie_id=movie_id)
+    if cinema_hall_id:
+        update_ms.update(cinema_hall_id=cinema_hall_id)
```

Comment:
The `update()` method is called multiple times, which is inefficient. Consider combining the updates into a single call to `update()` with a dictionary of fields to update, e.g., `update(show_time=show_time, movie_id=movie_id, cinema_hall_id=cinema_hall_id)`.